editoast/src/views/train_schedule/mod.rs

```diff
@@ -0,0 +1,833 @@
+
+    let infra = scenario.infra_id.expect("Scenario should have an infra id");
+
+    let mut res = Vec::new();
+    for schedule in schedules {
+        let sim_report = create_simulation_report(
+            infra,
+            schedule,
+            &projection,
+            &projection_path_payload,
+            db_pool.clone(),
+            core.as_ref(),
+        )
+        .await?;
+        res.push(sim_report);
+    }
+
+    Ok(Json(res))
+}
+
+#[post("/standalone_simulation/")]
+async fn standalone_simulation(
+    db_pool: Data<DbPool>,
+    train_schedules: Json<Vec<TrainSchedule>>,
+    core: Data<CoreClient>,
+) -> Result<Json<Vec<i64>>> {
+    let train_schedules = train_schedules.into_inner();
+
+    let id_timetable = train_schedules[0].timetable_id;
+
+    let timetable = Timetable::retrieve(db_pool.clone(), id_timetable)
+        .await?
+        .ok_or(TrainScheduleError::TimetableNotFound {
+            timetable_id: id_timetable,
+        })?;
+
+    let scenario = scenario_from_timetable(&timetable, db_pool.clone())?;
+
+    let request_payload =
+        create_backend_request_payload(&train_schedules, &scenario, db_pool.clone()).await?;
+    let response_payload = run_simulation(request_payload, core.as_ref()).await?;
+    let simulation_outputs = process_simulation_response(response_payload);
+
+    assert!(train_schedules.len() == simulation_outputs.len());
+
+    let mut res_ids = Vec::new();
+    let mut conn = db_pool.get()?;
+    // Start a transaction
+    conn.transaction::<_, InternalError, _>(|conn| {
+        // Save inputs
+        for train_schedule in train_schedules {
+            let id = train_schedule
+                .create_conn(conn)?
+                .id
+                .expect("Train schedule should have an id");
+            res_ids.push(id);
+        }
+        // Save outputs
+        for (i, mut simulation_output) in simulation_outputs.into_iter().enumerate() {
+            simulation_output.train_schedule_id = Some(Some(res_ids[i]));
+            simulation_output.create_conn(conn)?;
+        }
+        Ok(())
+    })?;
+    Ok(Json(res_ids))
+}
+
+#[derive(Debug, Serialize)]
+struct TrainPath {
+    pub route_paths: Vec<RoutePath>,
+}
+
+impl From<Pathfinding> for TrainPath {
+    fn from(value: Pathfinding) -> Self {
+        TrainPath {
+            route_paths: value.payload.route_paths.to_owned(),
+        }
+    }
+}
+
+impl From<&PathfindingPayload> for TrainPath {
+    fn from(value: &PathfindingPayload) -> Self {
+        TrainPath {
+            route_paths: value.route_paths.to_owned(),
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+struct SimulationRequest {
+    pub infra: i64,
+    pub rolling_stocks: Vec<RollingStock>,
+    pub train_schedules: Vec<TrainSchedule>,
+    pub electrical_profile_set: Option<String>,
+    pub trains_path: TrainPath,
+}
+
+impl AsCoreRequest<SimulationResponse> for SimulationRequest {
```

Comment:
If this PR is merged after #4046 you'll need to change this line:

```suggestion
impl AsCoreRequest<Json<SimulationResponse>> for SimulationRequest {
```

(a fix to allow non-json responses, possibly empty)